infrastructure/src/main/java/org/corfudb/infrastructure/ServerContext.java

```diff
@@ -144,7 +143,7 @@ public class ServerContext implements AutoCloseable {
     private final EventLoopGroup clientGroup;
 
     @Getter
-    private final EventLoopGroup workerGroup;
+    private EventLoopGroup workerGroup;
```

Comment:
This needs to be volatile. 
What's more, since `final` has to be removed, we need to be careful of race condition. `ServerContext::close` might still be fine (resource leak), but `CorfuServerNode::start` binding to a closed worker group can lead to disconnection errors. Seems like you could just avoid this race by checking `bindFuture.channel().active() && httpServerFuture.channel().active()` in the beginning of  `refreshWorkerGroupThreads()`.